spark3-extensions/src/main/antlr/org.apache.spark.sql.catalyst.parser.extensions/IcebergSqlExtensions.g4

```diff
@@ -83,6 +83,7 @@ constant
     : number                          #numericLiteral
     | booleanValue                    #booleanLiteral
     | STRING+                         #stringLiteral
+    | identifier STRING               #typeConstructor
```

Comment:
Is this can support calc expr like `CURRENT_TIMESTAMP- INTEVER '7' DAYS`?

Reply:
Raised this question in our slack channel.